### ISSUE TITLE
Alarm Control Panels not supported in 3.0.0?

### ISSUE THREAD

**Issue by jamieshaw** [home-assistant/homebridge-homeassistant#245]
Since upgrading to 3.0.0; my Alarm Control Panel has disappeared from my HomeKit setup.
Removing the persist file and reading all accessories has not fixed the issue — is this a known bug?

**Comment by robbiet480** [comment]
No changes were made in 3.0.0. The last change was made in 2.3.0 by @schmittx to add support for alarm panels with codes. @schmittx can provide better support than I will be able to for this...

**Comment by jamieshaw** [comment]
Interesting.  Just tested by downgrading to 2.3.1 and the alarm popped back up.

Are you seeing this issue on your end also?

**Comment by robbiet480** [comment]
Sorry, I don't have any alarm control panels! I believe @schmittx does though which is why I looped him in

**Comment by jamieshaw** [comment]
Fair enough. Thanks, hopefully a simple enough fix. 👍🏻

**Comment by schmittx** [comment]
@jamieshaw Strange...mine has been working fine, no issues here. The alarm_control_file file hasn't been changed since 2.3.1 except for ESlint fixes...

**Comment by jamieshaw** [comment]
@schmittx Definitely strange. Just attempted to upgrade back to the latest and the alarm disappears once Homebridge is reset.  Downgraded back to 2.3.1 and it's popped straight back up — definitely seeing an issue here.

I'm using the MQTT Alarm Control Panel, just for reference, I'm not sure that should make any difference to Homebridge though?

**Comment by schmittx** [comment]
@jamieshaw Nope, shouldn't make a different. Are you sure it's not just being moved to another room within HomeKit (i.e. "Default Room" or "Home")?

**Comment by jamieshaw** [comment]
@schmittx Nope; completely removed. Normally new devices get added to "Office" where I've placed the 'Hub' in the HomeKit config, and also end up at the bottom of the favourites.  

No such luck here, just completely removed from HomeKit altogether.

**Comment by schmittx** [comment]
OK, definitely weird.

I'll try to dig into some night this week and see if I can figure out what's going on...

**Comment by jamieshaw** [comment]
Cool. If there is anything I can do to help, let me know.

No issues seem to be flagged in Homebridge's logs. I'm unsure if there are any other logs generated by Homebridge or similar that might be of assistance…

**Comment by schmittx** [comment]
Can you paste your Homebridge config (just for `homebridge-homeassistant`) as well as the entity details for your alarm control panel (name, attributes, etc.)?

**Comment by jamieshaw** [comment]
**Homebridge config:**
```
{
	"bridge": {
		"name": "~~REDACTED~~",
		"model": "Mac mini (Late 2012)",
		"manufacturer": "Apple",
		"serialNumber": "~~REDACTED~~",
		"username": "~~REDACTED~~",
		"port": 51826,
		"pin": "~~REDACTED~~"
	},

	"platforms": [{
		"platform": "Server",
		"name": "~~REDACTED~~",
		"port": 51827,
		"log" : "/Users/~~REDACTED~~/.homebridge/homebridge.log",
		"restart": "pm2 restart homebridge"
	},{
		"platform": "Info",
		"name": "Homebridge Info Server",
		"port": 51828,
		"updateFrequency": 15000,
		"updateCheckFrequency": 3600000
	},{
		"platform": "websocket",
		"name": "~~REDACTED~~",
		"port": 51829
	},{
		"platform": "HomeAssistant",
		"name": "Home Assistant",
		"host": "https://127.0.0.1:51825",
		"password": "~~REDACTED~~",
		"default_visibility": "visible",
		"logging": false,
		"verify_ssl": false
	},{
		"platform": "Nest",
		"token": "~~REDACTED~~",
		"clientId": "~~REDACTED~~",
		"clientSecret": "~~REDACTED~~",
		"code": "~~REDACTED~~"
	},{
		"platform": "Camera-ffmpeg",
		"cameras": [{
		"name": "Front Door",
		"videoConfig": {
			"source": "-rtsp_transport tcp -i rtsp://~~REDACTED~~",
			"stillImageSource": "-i ~~REDACTED~~ -vframes 1 -r 1",
			"maxStreams": 4,
			"maxWidth": 1280,
			"maxHeight": 720,
			"maxFPS":24,
			"vcodec": "h264_videotoolbox"
			}
		}]
	}],

	"accessories": [{
		"accessory": "mqtt-switch-tasmota",
		"switchType": "other",
		"name": "~~REDACTED~~",
		"url": "mqtt://~~REDACTED~~",
		"username": "~~REDACTED~~",
		"password": "~~REDACTED~~",
		"topics": {
			"statusGet": "~~REDACTED~~",
			"statusSet": "~~REDACTED~~"
		},
		"onValue": "on",
		"offValue": "off",
		"retain" : true,
		"qos" : 1,
		"manufacturer": "Homebridge",
		"model": "Switch",
		"serialNumberMAC": ""
	},{
		"accessory": "mqtt-switch-tasmota",
		"switchType": "other",
		"name": "~~REDACTED~~",
		"url": "mqtt://~~REDACTED~~",
		"username": "~~REDACTED~~",
		"password": "~~REDACTED~~",
		"topics": {
			"statusGet": "~~REDACTED~~",
			"statusSet": "~~REDACTED~~"
		},
		"onValue": "on",
		"offValue": "off",
		"retain" : true,
		"qos" : 1,
		"manufacturer": "Homebridge",
		"model": "Switch",
		"serialNumberMAC": ""
	}]
}
```

**Alarm Control Panel Config**
```
alarm_control_panel:
  - name: "Security Alarm"
    platform: manual_mqtt
    code: !secret security_alarm_code
    state_topic: ~~REDACTED~~
    command_topic: ~~REDACTED~~
    payload_disarm: "disarm"
    payload_arm_home: "arm_home"
    payload_arm_away: "arm_away"
    pending_time: 0
    triggered:
        pending_time: 30
    trigger_time: 300
    disarm_after_trigger: False
    qos: 2
```

**Customize**:
```
alarm_control_panel.security_alarm:
    friendly_name: "Alarm"
    homebridge_alarm_code: !secret security_alarm_code
```

**Customize Glob**:
```
"*security*":
    icon: mdi:shield-half-full
```

**Comment by jamieshaw** [comment]
Fixed it; and I feel foolish – Just had a look at the code and it's a simple fix, tweaked my install and it's working fine.

Look as though, in index.js on line 27, `alarm_control_panel` was mistakenly removed for `automation`. Just needs adding back in *facepalm*.

I would do it myself but I'm not quite used to GitHub. If someone cares to explain, I'll happily give it a go, otherwise, feel free to quickly fix yourself. 🙂 👍🏻

**Comment by schmittx** [comment]
@jamieshaw Good catch, I think that was my fault with some of the recent updates, my apologies! I didn't catch it because I manually specify the `supported_types` in my Homebridge config. PR is submitted, just need @robbiet480 to approve, merge, and update.

**Comment by robbiet480** [comment]
Thanks! Fix published as 3.0.2.
